cmd/shortener/main.go

```diff
@@ -67,16 +77,75 @@ func handlerGet(store *storage.MapStorage) http.HandlerFunc {
 	}
 }
 
+func PostShortenRequest(fileStorage *storage.FileStorage) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var originURL models.OriginalURL
+
+		if r.Method != http.MethodPost {
+			http.Error(w, "Bad Request", http.StatusMethodNotAllowed)
+			return
+		}
+
+		if err := json.NewDecoder(r.Body).Decode(&originURL); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		url := originURL.URL
+		shortID := utils.ShortenURL(url)
+		shortLink := models.ShortLink{UUID: uuid.NewString(),
+			ShortURL:    shortID,
+			OriginalURL: url,
+		}
+
+		fileStorage.SaveShortLink(shortLink)
+		shortURL := config.BaseURL + shortID
+
+		resp := models.ShortURL{
+			Result: shortURL,
+		}
+
+		response, err := json.MarshalIndent(resp, "", "   ")
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusCreated)
+		w.Write(response)
+	}
+}
+
 func main() {
 	config.Init()
 
+	run()
+
 	store := storage.NewMapStorage()
+	fileStorage := storage.NewFileStorage(config.FileStorage, store)
+
+	err := fileStorage.LoadFromFile()
+	if err != nil {
+		logger.Log.Error("Store not load")
+	}
 
 	r := chi.NewRouter()
 	r.Route("/", func(r chi.Router) {
-		r.Post("/", handler(store))
-		r.Get("/{id}", handlerGet(store))
+		r.Post("/", logger.RequestLogger(compress.GzipCompress(handler(fileStorage))))
+		r.Get("/{id}", logger.RequestLogger(compress.GzipCompress(handlerGet(store))))
+		r.Route("/api/", func(r chi.Router) {
+			r.Post("/shorten", logger.RequestLogger(PostShortenRequest(fileStorage)))
+		})
 	})
 
 	log.Fatal(http.ListenAndServe(config.Address, r))
 }
+
+func run() {
+	if err := logger.Initialize(config.LogLevel); err != nil {
+		logger.Log.Error(err.Error())
+	}
+
+	logger.Log.Info("Running server on", zap.String("Address", config.Address))
+}
```

Comment:
Во-первых из этого метода стоит возвращать ошибку, так как в текущей реализации он логирует ошибку и выполнение продолжается, в то время как если мы не смогли инициализировать логгер, как минимум мы не должны его использовать? Во-вторых не совсем понятно назначение этого метода - он только инициализирует логгер? По его названию кажется, что его назначение запускать сервер.
```suggestion
func run() error {
	if err := logger.Initialize(config.LogLevel); err != nil {
		return error
	}

	logger.Log.Info("Running server on", zap.String("Address", config.Address))
}
```

Reply:
Исправил